addon/serializers/application.js

```diff
@@ -51,182 +51,207 @@ export default DS.RESTSerializer.extend({
+         * Here we set the links property to a serialized version
+         * of key and className. This info will be passed to
+         * adapter.findHasMany where we can deserialize to create
+         * the needed Parse query.
+         */
+        if (hash[key] && 'hasMany' === relationship.kind) {
+          if (!hash[key].__op && hash[key].__op !== 'AddRelation') {
+            if (!hash.links) {
+              hash.links = {};
+            }
+            hash.links[key] = JSON.stringify({
+              key: key,
+              className: hash[key].className
             });
           }
-        }
-      }
-    }, this );
-
-    this._super( type, hash );
-  },
 
-  serializeIntoHash: function( hash, type, snapshot, options ) {
-    Ember.merge( hash, this.serialize( snapshot, options ) );
+          delete hash[key].__type;
+          delete hash[key].className;
+          hash[key] = [];
+        }
+      }, this);
+    }
   },
 
-  serializeAttribute: function( snapshot, json, key, attribute ) {
+  serializeAttribute(record, json, key, attribute) {
     // These are Parse reserved properties and we won't send them.
-    if ( 'createdAt' === key ||
-         'updatedAt' === key ||
-         'emailVerified' === key ||
-         'sessionToken' === key
+    if ('createdAt' === key ||
+        'updatedAt' === key ||
+        'emailVerified' === key ||
+        'sessionToken' === key ||
+        'password' === key
     ) {
       delete json[key];
 
     } else {
-      this._super( snapshot, json, key, attribute );
+      this._super(record, json, key, attribute);
     }
   },
 
-  serializeBelongsTo: function( snapshot, json, relationship ) {
-    var key         = relationship.key,
-        belongsToId = snapshot.belongsTo(key, { id: true });
+  serializeBelongsTo(record, json, relationship) {
+    var key = relationship.key,
+        belongsTo = record.belongsTo(key);
+
+    if (belongsTo) {
+      // @TODO: Perhaps this is working around a bug in Ember-Data? Why should
+      // promises be returned here.
+      if (belongsTo instanceof DS.PromiseObject) {
+        if (!belongsTo.get('isFulfilled')) {
+          throw new Error('belongsTo values *must* be fulfilled before attempting to serialize them');
+        }
+
+        belongsTo = belongsTo.get('content');
+      }
+
+      var _className = this.parseClassName(belongsTo.type.modelName);
+
+      if (_className === 'User') {
+        _className = '_User';
+      }
 
-    if ( belongsToId ) {
       json[key] = {
-        '__type'    : 'Pointer',
-        'className' : this.parseClassName(key),
-        'objectId'  : belongsToId
+        '__type': 'Pointer',
+        'className': _className,
+        'objectId': belongsTo.id
       };
+
     }
   },
 
-  parseClassName: function( key ) {
-    if ( 'parseUser' === key) {
+  serializeIntoHash(hash, type, snapshot, options) {
+    var ParseACL = snapshot.record.get('ParseACL');
+
+    // Add ACL
```

Comment:
This seems to warrant some documentation, or a link to Parse's documentation on why an ACL is needed here.